### PR TITLE
feat: Support spaces between field name and value

### PR DESCRIPTION
**Pull request by Darkheir**
Allow optional spaces between the field name and the value the same way Lucene allows them

### CODE REVIEW

**Comment by fulmicoton** [comment]
@trinity-1686a can you review? Does it looks sound?

File: query-grammar/src/query_grammar.rs [hunk 1]
```diff
@@ -1283,6 +1283,10 @@ mod test {
             super::field_name("~my~field:a"),
             Ok(("a", "~my~field".to_string()))
         );
+        assert_eq!(
+            super::field_name(".my.field.name : a"),
```
Comment (trinity-1686a) [inline query-grammar/src/query_grammar.rs:1287]: can you also add a `test_parse_query_to_ast_helper` test somewhere to make sure this works (and stays working) for both the strict and the lenient parser

**Review by trinity-1686a — APPROVED** [review]
lgtm, afaict it doesn't introduce any problematic ambiguity
it makes redundant some of the multisplace handling in the lenient parser, but i'll clean that up myself eventually

**Comment by Darkheir** [comment]
Thank you for the review :)

I added few tests, but don't hesitate to ask for more if you see some edge cases that could not be covered by the existing ones.

**Comment by trinity-1686a** [comment]
thanks!
